include/kernel.h

```diff
@@ -5356,6 +5371,26 @@ __syscall void k_str_out(char *c, size_t n);
  */
 __syscall int k_float_disable(struct k_thread *thread);
 
+#ifdef CONFIG_THREAD_RUNTIME_STATS
+/**
+ * @brief Return the total execution cycles of a thread
+ *
+ * @param thread ID of thread.
+ * @return The total execution cycles
+ */
+static inline uint64_t k_thread_total_exec_cycles_get(struct k_thread *thread)
```

Comment:
ugly name, maybe k_thread_runtime_stats_get()?

Reply:
My assumption is that the runtime stats will be expanded in the future to gather more metrics. So the get function needs to be specified on what it's returning. Let me think of another name.

Reply:
maybe put the data in a struct and return all information available, so we do not need to create an API for every new metric

Reply:
Done.